### PR TITLE
All icon svgs now have a default width/height

### LINKED ISSUE #7922: Flash of huge notebook icon on open
## Description

JupyterLab 2.0.0rc2 flashes a huge Notebook icon when first opening in the launcher page, before it is immediately resized to a small icon on the launcher next to "Notebook"

## Reproduce

1. Install JupyterLab 2.0.0rc2
2. Open the page
3. Note the flash of a huge icon, which immediately scales down to be the Notebook icon in the launcher

## Expected behavior

No flash of a huge icon

## Context

- Operating System and version: macOS High Sierra
- Browser and version: Firefox 73
- JupyterLab version: 2.0.0rc2

CC @telamonian, @tgeorgeux

### PR DESCRIPTION
**Pull request by telamonian**
## References

fixes #7922

All icons now have a default width/height defined inline. For most icons this is 24x24

## Code changes

NA

## User-facing changes

Firefox: less noticable FOUC on page reload for icons rendered using react

## Backwards-incompatible changes

NA

### CODE REVIEW

File: packages/ui-components/style/icons/toolbar/bug.svg [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<svg viewBox="0 0 24 24" height="18" width="18" xmlns="http://www.w3.org/2000/svg">
+<svg viewBox="0 0 24 24" height="24" width="24" xmlns="http://www.w3.org/2000/svg">
```
Comment (jasongrout) [inline packages/ui-components/style/icons/toolbar/bug.svg]: Should all the toolbar icons stay at 18x18?

**Comment by jasongrout** [comment]
I'm also poking around the UX to see if I notice anything off.

As a side note - perhaps we should just specify one dimension (e.g., height of 24), and let the viewbox coordinates set the intrinsic aspect ratio. See https://www.w3.org/TR/SVGMobile12/coords.html#IntrinsicSizing

**Comment by jasongrout** [comment]
Having poked around a bit, I think:

* Set one of height or width, and let the viewbox dimensions control the intrinsic aspect ratio. I recommend setting height, as that is usually the more constrained dimension (e.g., on a toolbar)
* Set the height of most things to 16px (which is the "normal" size in the UI): https://github.com/jupyterlab/jupyterlab/blob/30aef10f21707c0ec74ee6acc487a6c0948ca821/packages/ui-components/src/style/icon.ts#L69-L76 (I checked, and the toolbar icons and file browser icons are 16px). I think, like you mention, for FOUC, it's a better ux to be smaller and grow in size, rather than be bigger and shrink. Starting smaller and growing in size is a common UX paradigm for springing into existence, whereas starting bigger and shrinking looks weird to me. So that's why I recommend 16px for just about everything, rather than a larger size. I think one exception there would be the Jupyter logo, which is used in the splash screen?

**Comment by jasongrout** [comment]
And definitely +1 to having svgo make the svg formatting uniform!

**Comment by jasongrout** [comment]
>     * Set one of height or width, and let the viewbox dimensions control the intrinsic aspect ratio. I recommend setting height, as that is usually the more constrained dimension (e.g., on a toolbar)

By the way, the reason for this is (I think) setting the width and height with a different aspect ratio from the viewBox leads to a distorted image.

**Comment by telamonian** [comment]
> And definitely +1 to having svgo make the svg formatting uniform!

Yeah, I'd like to get our svg formatting/attributes uniform once, automate the process, and then never again have to do a manual pass over our svg collection. Unfortunately, svgo still requires significant manual supervision. The problem is that by default svgo is set up for using svgs in `` tags, and not for our use case, which is inline svg + CSS stylesheet.

For example, svgo removes `fill='#000'` from any svg element, since that's the default value. However, the `fill` attr needs to be present in order to be able to change the fill color using CSS, so right now svgo will break eg our `html5Icon`. I'm experimenting with different configs, but I'm not sure that svgo can get us where we want to be.

I had better luck with [svgcleaner](https://github.com/RazrFalcon/svgcleaner), but that's a rust project and so cannot be nicely integrated with our dev tools. That's not an issue if it's just me using it, but obviously it would be nice if adding svgs was easy for other contributors as well

**Comment by telamonian** [comment]
> Starting smaller and growing in size is a common UX paradigm for springing into existence, whereas starting bigger and shrinking looks weird to me.

I was thinking it was best to split the difference between our smallest and largest common icon use cases. But if grow-to-size is the paradigm we want, then I agree that the default size should be 16px.

**Comment by jasongrout** [comment]
> But if grow-to-size is the paradigm we want

Just to be clear: I was only voicing my thoughts, not trying to set what "we" want. Do you agree? I don't have any real evidence against shrink-to-size, other than (very subjectively, of course) it looks weird to me.

**Comment by blink1073** [comment]
+1 for 16px and the grow-to-size behavior.

File: packages/ui-components/style/icons/toolbar/bug.svg (hunk 1 above)
Comment (jasongrout) [reply packages/ui-components/style/icons/toolbar/bug.svg]: Other discussion suggests that we should do 16px instead.

**Comment by telamonian** [comment]
Oh no, you merged too soon! There was one last fix to the command palette icon (its stylesheet was getting clobbered)

**Comment by telamonian** [comment]
I moved the final commit to a new PR: #7938

**Pull request merged**